x-pack/plugins/aiops/public/application/utils/query_utils.test.ts

```diff
@@ -0,0 +1,254 @@
+};
+
+const selectedGroupMock: GroupTableItem = {
+  id: '21289599',
+  docCount: 20468,
+  pValue: 2.2250738585072626e-308,
+  group: {
+    'error.message': 'rate limit exceeded',
+    message: 'too many requests',
+    'user_agent.original.keyword': 'Mozilla/5.0',
+  },
+  repeatedValues: {
+    'beat.hostname.keyword': 'ip-192-168-1-1',
+    'beat.name.keyword': 'i-1234',
+    'docker.container.id.keyword': 'asdf',
+  },
+  histogram: [],
+};
+
+describe('query_utils', () => {
+  describe('buildBaseFilterCriteria', () => {
+    it('returns range filter based on minimum supplied arguments', () => {
+      const baseFilterCriteria = buildBaseFilterCriteria('the-time-field-name', 1234, 5678);
+
+      expect(baseFilterCriteria).toEqual([
+        {
+          range: {
+            'the-time-field-name': {
+              format: 'epoch_millis',
+              gte: 1234,
+              lte: 5678,
+            },
+          },
+        },
+      ]);
+    });
+
+    it('returns filters including default query with supplied arguments provided via UI', () => {
+      const baseFilterCriteria = buildBaseFilterCriteria(
+        '@timestamp',
+        1640082000012,
+        1640103600906,
+        { match_all: {} }
+      );
+
+      expect(baseFilterCriteria).toEqual([
+        {
+          range: {
+            '@timestamp': {
+              format: 'epoch_millis',
+              gte: 1640082000012,
+              lte: 1640103600906,
+            },
+          },
+        },
+        { match_all: {} },
+      ]);
+    });
+
+    it('includes a term filter when including a selectedChangePoint', () => {
+      const baseFilterCriteria = buildBaseFilterCriteria(
+        '@timestamp',
+        1640082000012,
+        1640103600906,
+        { match_all: {} },
+        selectedChangePointMock
+      );
+
+      expect(baseFilterCriteria).toEqual([
+        {
+          range: {
+            '@timestamp': {
+              format: 'epoch_millis',
+              gte: 1640082000012,
+              lte: 1640103600906,
+            },
+          },
+        },
+        { match_all: {} },
+        { term: { 'meta.cloud.instance_id.keyword': '1234' } },
+      ]);
+    });
+
+    it('includes a term filter with must_not when excluding a selectedChangePoint', () => {
+      const baseFilterCriteria = buildBaseFilterCriteria(
+        '@timestamp',
+        1640082000012,
+        1640103600906,
+        { match_all: {} },
+        selectedChangePointMock,
+        false
+      );
+
+      expect(baseFilterCriteria).toEqual([
+        {
+          range: {
+            '@timestamp': {
+              format: 'epoch_millis',
+              gte: 1640082000012,
+              lte: 1640103600906,
+            },
+          },
+        },
+        { match_all: {} },
+        { bool: { must_not: [{ term: { 'meta.cloud.instance_id.keyword': '1234' } }] } },
+      ]);
+    });
+
+    it('includes multiple term filters when including a selectedGroupMock', () => {
```

Comment:
nit: do we have to specify the `Mock` part in the test name?

Reply:
Fixed in [5f8e186](https://github.com/elastic/kibana/pull/141213/commits/5f8e186dab30d5e33613428143a4034b1214154d).